### PR TITLE
ARROW-442: [Python] Inspect Parquet file metadata from Python

### PR DESCRIPTION
**Pull request by wesm**
I also made the Cython parquet extension "private" so that higher level logic (e.g. upcoming handling of multiple files) can be handled in pure Python (which doesn't need to be compiled)

Requires PARQUET-828 for the test suite to pass.

### CODE REVIEW

**Comment by wesm** [comment]
@xhochy I put this up before being fully done, also to let you know I'd done some refactoring in the Parquet Cython extension

**Comment by wesm** [comment]
Removed the WIP -- @xhochy take a look when you can. thanks

File: python/pyarrow/_parquet.pyx [hunk 1]
```diff
@@ -0,0 +1,520 @@
...
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# cython: profile=False
+# distutils: language = c++
+# cython: embedsignature = True
+
+from pyarrow._parquet cimport *
```
Comment (xhochy) [inline python/pyarrow/_parquet.pyx]: This should not be needed?

**Review by xhochy — APPROVED** [review]
+1, LGTM

**Pull request closed without merging**